### DISCUSSION TITLE
chmod: Unable to change file on.... (Need help with installation / clean system)?

### DISCUSSION THREAD

**Discussion by ThomasMatt** [Homebrew/discussions#797] [Everyday usage]
Hello all, Thank you for popping by.

I'm experiencing a few troubles with terminal and homebrew now it's not installing..
Please see the image I have linked. I am mainly getting this code: 

`Press RETURN to continue or any other key to abort
==> /usr/bin/sudo /bin/chmod u+rwx /usr/local/bin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/sbin /usr/local/share /usr/local/opt /usr/local/var /usr/local/Frameworks /usr/local/etc/bash_completion.d /usr/local/lib/pkgconfig /usr/local/share/aclocal /usr/local/share/doc /usr/local/share/info /usr/local/share/locale /usr/local/share/man /usr/local/share/man/man1 /usr/local/share/man/man3 /usr/local/share/man/man5 /usr/local/share/man/man8 /usr/local/share/zsh /usr/local/share/zsh/site-functions
==> /usr/bin/sudo /bin/chmod g+rwx /usr/local/bin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/sbin /usr/local/share /usr/local/opt /usr/local/var /usr/local/Frameworks /usr/local/etc/bash_completion.d /usr/local/lib/pkgconfig /usr/local/share/aclocal /usr/local/share/doc /usr/local/share/info /usr/local/share/locale /usr/local/share/man /usr/local/share/man/man1 /usr/local/share/man/man3 /usr/local/share/man/man5 /usr/local/share/man/man8
==> /usr/bin/sudo /bin/chmod 755 /usr/local/share/zsh /usr/local/share/zsh/site-functions
chmod: Unable to change file mode on /usr/local/share/zsh: Operation not permitted
chmod: Unable to change file mode on /usr/local/share/zsh/site-functions: Operation not permitted
Failed during: /usr/bin/sudo /bin/chmod 755 /usr/local/share/zsh /usr/local/share/zsh/site-functions
`

I have tried adding Bash to the privacy full disk access, i've also given terminal full disk access.
Any help would be appreciated.

**Comment by gromgit** [comment]
_Sidenote:_ When inserting multiple lines of code, either use the GitHub editor's "insert code" (`< >`) button, or manually enclose the lines in _triple_ backticks. It looks like you used single backticks above, which is meant for _inline_ code inclusions. 
> ```
> ==> /usr/bin/sudo /bin/chmod 755 /usr/local/share/zsh /usr/local/share/zsh/site-functions
> chmod: Unable to change file mode on /usr/local/share/zsh: Operation not permitted
> chmod: Unable to change file mode on /usr/local/share/zsh/site-functions: Operation not permitted
> Failed during: /usr/bin/sudo /bin/chmod 755 /usr/local/share/zsh /usr/local/share/zsh/site-functions
> ```

What's the output of `/bin/ls -ld@eO /usr/local/share /usr/local/share/zsh /usr/local/share/zsh/site-functions`?

**Reply by ThomasMatt** [reply]
`
➜  ~ /bin/ls -ld@eO /usr/local/share /usr/local/share/zsh /usr/local/share/zsh/site-functions
drwxrwxrwx  14 admin  admin  restricted 448 13 Feb 03:11 /usr/local/share
drwxrwxrwx   3 admin  admin  restricted  96 19 Nov  2019 /usr/local/share/zsh
drwxrwxrwx   2 admin  admin  restricted  64 13 Feb 14:35 /usr/local/share/zsh/site-functions
➜  ~ 
`

**Reply by gromgit** [reply]
Again, please use triple backticks (```) or the GitHub editor's "insert code" button to format multiline output correctly. Everything you've posted so far is borderline unreadable.
> ```
> drwxrwxrwx 14 admin admin restricted 448 13 Feb 03:11 /usr/local/share
> drwxrwxrwx 3  admin admin restricted  96 19 Nov 2019  /usr/local/share/zsh 
> drwxrwxrwx 2  admin admin restricted  64 13 Feb 14:35 /usr/local/share/zsh/site-functions
> ```

The `restricted` flags should not be there, since (IIRC) SIP (System Integrity Protection) specifically does _not_ touch anything under `/usr/local`.

Just to confirm, what's the output of:
```
ls -ld@eO /usr/local
csrutil status
```

Also:
> I have tried adding Bash to the privacy full disk access, i've also given terminal full disk access.

Try rebooting your Mac, then logging in again and double-check that the above settings are still set, then open a Terminal and do the `ls` commands above again. If you still see `restricted`, the only thing else I can think of is to disable SIP, which is definitely _not_ recommended.
